day_one/src/main.rs

```diff
@@ -0,0 +1,47 @@
+use std::fs;
+fn main() {
+    let raw_input = fs::read_to_string("/Users/phil/Developer/aoc2024/day_one/src/input.txt")
+        .expect("Should have been able to read the file");
+
+    let [list_one, list_two] = parse_and_sort_lists(&raw_input);
+
+    let mut total_distance: i32 = 0;
+
+    for index in 0..list_one.len() {
+        total_distance += (list_one[index] - list_two[index]).abs();
+    }
+
+    println!("{total_distance}");
+
+    let mut total_similarity_scores: i32 = 0;
+    for list_value in list_one {
+        total_similarity_scores += get_similarity_score(list_value, &list_two)
+    }
+    println!("{total_similarity_scores}")
```

Comment:
It's personal preference at the end of the day but if you want a one-liner using iterators, you can do something like this:

```rust
    let total_similarity_scores = list_one
        .iter()
        .map(|n| get_similarity_score(*n, &list_two))
        .sum::<i32>();
```